trieste/acquisition/sampler.py

```diff
@@ -276,3 +279,190 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
+        """
+        Return a function that evaluates a particular sample at a set of `N` query
+        points (each of dimension `D`) i.e. takes input of shape `[N, D]` and returns
+        shape `[N, 1]`.
+
+        :return: Queryable function representing a sample.
+        """
+
+
+class RandomFourierFeatureThompsonSampler(ContinuousSampler):
+    r"""
+
+    This class builds functions that approximate samples from an underlying Gaussian
+    process model. For tractibility, the Gaussian process is approximated with a Bayesian
+    Linear model across a set of features sampled from the Fourier feature decomposition of
+    the model's kernel. See :cite:`hernandez2014predictive` for details.
+
+    A key property of these functions is that the same function draw (sample) is evaluated
+    for all queries. This property is known as consistency. Achieving consistency for exact
+    sample draws from a  GP is prohibitevly costly because it scales cubically with the number
+    of query points. However, finite feature representations can be evaluated with constant cost
+    regardless of the required number of queries.
+
+    In particular, we approximate the Gaussian processes' posterior samples as the finite feature
+    approximation
+
+    .. math:: \hat{f}(x) = \sum_{i=1}^m \phi_i(x)\theta_i
+
+    where :math:`\phi_i` are m Fourier features and :math:`\theta_i` are
+    feature weights sampled from a posterior distribution that depends on the feature values at the
+    model's datapoints.
+
+    Our implementation follows :cite:`hernandez2014predictive`, with our calculations
+    differing slightly depending on properties of the problem, in particular,  with respect to the
+    number of considered features m and the number of data points n.
+
+    If :math:`m<n` then we follow Appendix A of :cite:`hernandez2014predictive` and calculate the
+    posterior distribution for :math:`\theta` following their Bayesian linear regression motivation,
+    i.e. the computation revolves around an O(m^3)  inversion of a design matrix.
+
+    If :math:`n<m` then we use the kernel trick to recast computation to revolve around an O(n^3)
+    inversion of a gram matrix. As well as being more efficient in early BO
+    steps (where :math:`n<m`), this second computation method allows must larger choices
+    of m (as required to approximate very flexible kernels).
+
+    """
+
+    def __init__(self, dataset: Dataset, model: GPflowPredictor, num_features: int = 1000):
+        """
+        :param dataset: The data from the observer. Must be populated.
+        :param model: The model to sample from.
+        :param num_features: The number of features used to approximate the kernel.
+        :raise ValueError: If ``dataset`` is empty.
+        """
+        super().__init__(dataset, model)
+        self._num_features = num_features  # [m]
+        self._num_data = len(self._dataset.query_points)  # [n]
```

Comment:
This comments are to connect the variable names to the terms of the equations in the class docstring? The square brackets suggest to me that they are shape annotations.